### ISSUE TITLE
Can't drag email icon in Navigation (Current Folder) bar to a folder / QuickFolder drop target

### ISSUE THREAD

**Issue by PeterW-gif** [RealRaven2000/QuickFolders#84]
TB 78.5.1 quickfolders V5.0 In email display tab the feature to drag the 'current email to a Quickfolders tab; doesn't work now (used to under previous version)
The hand icon shows when the cursor is placed on the mail 'envelope' icon and changed to grabbed when clicked but the mail doesn't move when dragged to any folder in the folders tab toolbar
This under Windows 7 but also the same on my Macbook under OS 11

**Comment by PeterW-gif** [comment]
Quite right - that's what I meant - glad to see it wasn't just something I was doing wrong

**Comment by RealRaven2000** [comment]
> Quite right - that's what I meant - glad to see it wasn't just something I was doing wrong

The good news is I get an exception in Javascript error console when I start to drag, so that's an easy enough pointer for the rewrite:

```
Uncaught ReferenceError: TransferData is not defined
    msgObs_startDrag chrome://quickfolders/content/quickfolders.js:1468
    ondragstart chrome://messenger/content/messenger.xhtml:1

```

**Comment by RealRaven2000** [comment]
..the fix was relatively trivial, but I noticed another problem - when multiple emails are selected (e.g. a whole thread) the Navigation Bar (a.k.a. Current Folder Bar) is hidden. That's makes it not useful, I really want to be able to select a bunch of mails and then drag the mail envelope. It means I probably need to move the current folder bar to a different container in the main document.

Fixed version, so far:

[QuickFolders-wx-5.1pre29.zip](https://github.com/RealRaven2000/QuickFolders/files/5647067/QuickFolders-wx-5.1pre29.zip)

**Comment by RealRaven2000** [comment]
> when multiple emails are selected (e.g. a whole thread) the Navigation Bar (a.k.a. Current Folder Bar) is hidden. That's makes it not useful, I really want to be able to select a bunch of mails and then drag the mail envelope. It means I probably need to move the current folder bar to a different container in the main document.

Fixed that one, too:
[QuickFolders-wx-5.1pre30.zip](https://github.com/RealRaven2000/QuickFolders/files/5647071/QuickFolders-wx-5.1pre30.zip)

**Comment by PeterW-gif** [comment]
Sorry, just downloaded that zip file but could not find any .xpi file inside

**Comment by RealRaven2000** [comment]
> Sorry, just downloaded that zip file but could not find any .xpi file inside

You can drag the zip file directly into Add-ons Manager. github doesn't allow uploading of xpi files, although internally they are the same.

**Comment by PeterW-gif** [comment]
Sorry didn't know that, just done it and your fix works - many thanks

**Comment by RealRaven2000** [comment]
fixed in release 5.1 on 10/Dec/2020
